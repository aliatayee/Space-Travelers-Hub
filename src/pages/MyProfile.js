import { React } from 'react';
import classes from './MyProfile.module.css';
import MyMission from './MyMission';

const MyProfile = () => (
  <div className={classes.myProfile}>
    <div>
      <MyMission />
    </div>
  </div>
);

export default MyProfile;
