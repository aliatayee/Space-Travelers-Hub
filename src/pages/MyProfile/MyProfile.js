import { React } from 'react';
import classes from '../Missions/MyProfile.module.css';
import MyMission from '../Missions/MyMission';
import MyRockets from '../../components/MyRockets/MyRockets';

const MyProfile = () => (
  <div className={classes.myProfile}>
    <div>
      <MyMission />
      <MyRockets />
    </div>
  </div>
);

export default MyProfile;
