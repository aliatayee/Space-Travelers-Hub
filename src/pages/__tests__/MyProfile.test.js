import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MyProfile from '../MyProfile/MyProfile';

describe('Test my profile', () => {
  it('Button component render', () => {
    const tree = renderer.create(<Provider store={store}><MyProfile /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Test My Rockets title', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.getByRole('heading', { name: /My Rockets/i })).toBeInTheDocument();
  });
  it('Test My Missions title', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.getByRole('heading', { name: /My Missions:/i })).toBeInTheDocument();
  });
});
