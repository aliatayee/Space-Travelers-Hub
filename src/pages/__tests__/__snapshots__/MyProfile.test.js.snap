// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my profile Button component render 1`] = `
<div
  className="grid grid-cols-2 m-4 gap-y-4 gap-x-8"
>
  <div
    className="border p-3 pb-5"
  >
    <h1
      className="font-bold text-100 border-b"
    >
      My Missions:
    </h1>
    <p>
      You have no missions
    </p>
  </div>
  <div
    className="border p-3 pb-5"
  >
    <h1
      className="font-bold text-100 border-b"
    >
      My Rockets
    </h1>
    <p>
      You reserved no rockets
    </p>
  </div>
</div>
`;
